pyvlx/pyvlx.py

```diff
@@ -47,6 +48,16 @@ async def connect(self):
         await login.do_api_call()
         if not login.success:
             raise PyVLXException("Login to KLF 200 failed, check credentials")
+        if self.connection.connection_counter & 1:
```

Comment:
I think this logic should be handled within HASS.

Reply:
I agree. At the moment it seems that this doesn't solve the problem at all. I still had a frozen KLF over the weekend even though I initiate a reboot via HASS at `async def on_hass_stop(event):`. So before HASS is stopped for a reboot.

However I'm not sure whether the KLF was also rebooting when I had rebooted HASS. Last time I only could observe one reboot, but at a timing when HASS was loaded, not when HASS was shutting down. I'll have to do some more tests with this.

Reply:
All approaches are somehow fishy ...

Reply:
I cleaned it up, so now only the reboot function is added and the connection counter, but no automatic reboot. On my custom component I have added an automatic reboot on hass stop event